validator/validate_api.py

```diff
@@ -72,21 +66,28 @@ def setup_parse_and_data(setup_state):
 
 def get_question_data_by_key(key, val):
     df = current_app.df
-    first_q = df["questions"][df["questions"][key] == val].iloc[0]
-    module_id = first_q.cvuid
-    uid = first_q.uid
-    has_numeric = df["questions"][df["questions"][key] == val].iloc[0].contains_number
+    # FIXME - should use all the questions and combine associated pages
+    # FIXME - last_q works better because of some dirty data getting through
+    # that has innovation pages but not the exact book those pages are from
```

Comment:
Hmm can you guarantee that `last_q` will be clean though?
Maybe it's better to do a select on questions with book data and possibly fallback to last if none have it?

Reply:
yes, moving the data to something that allows a join would help - and then we could control for getting the most-recent data, if the exact match isn't present.